fletcher/base.py

```diff
@@ -442,6 +481,8 @@ def astype(self, dtype, copy=True):
             raise NotImplementedError("Cast propagation in astype not yet implemented")
         else:
             dtype = np.dtype(dtype)
+            if pa.types.is_list(self.dtype.arrow_dtype) and dtype.kind == "U":
```

Comment:
Why only for Unicode?

Reply:
NumPy has implemented a different conversion for `list->unicode` than the default Python `__str__` method would do, the expectation is that the Python version is used.